DatadogSessionReplay/Sources/Recorder/Utilities/UIColor+SessionReplay.swift

```diff
@@ -0,0 +1,44 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-Present Datadog, Inc.
+ */
+
+#if os(iOS)
+
+import Foundation
+import UIKit
+import DatadogInternal
+
+extension UIColor: DatadogExtended {}
+
+private var identifierKey: UInt8 = 0
+
+extension DatadogExtension where ExtendedType: UIColor {
```

Comment:
Suggestion: I think we could make `UIColor` conform to `Hashable` here. This simplifies the process by removing the need for associated objects and leveraging Swift's native performance optimizations.